libs/test/SalesforceSDKTest/src/com/salesforce/androidsdk/accounts/UserAccountTest.java

```diff
@@ -167,30 +164,164 @@ public void testCreateAccountFromBundle() {
     public void testCreateAccountFromJSON() throws JSONException {
         JSONObject testJSON = createTestAccountJSON();
         UserAccount account = new UserAccount(testJSON);
-        Assert.assertEquals("Auth token should match", TEST_AUTH_TOKEN, account.getAuthToken());
-        Assert.assertEquals("Refresh token should match", TEST_REFRESH_TOKEN, account.getRefreshToken());
-        Assert.assertEquals("Login server URL should match", TEST_LOGIN_URL, account.getLoginServer());
-        Assert.assertEquals("Identity URL should match", TEST_IDENTITY_URL, account.getIdUrl());
-        Assert.assertEquals("Instance URL should match", TEST_INSTANCE_URL, account.getInstanceServer());
-        Assert.assertEquals("Org ID should match", TEST_ORG_ID, account.getOrgId());
-        Assert.assertEquals("User ID should match", TEST_USER_ID, account.getUserId());
-        Assert.assertEquals("User name should match", TEST_USERNAME, account.getUsername());
-        Assert.assertEquals("Community ID should match", TEST_COMMUNITY_ID, account.getCommunityId());
-        Assert.assertEquals("Community URL should match", TEST_COMMUNITY_URL, account.getCommunityUrl());
-        Assert.assertEquals("First name should match", TEST_FIRST_NAME, account.getFirstName());
-        Assert.assertEquals("Last name should match", TEST_LAST_NAME, account.getLastName());
-        Assert.assertEquals("Display name should match", TEST_DISPLAY_NAME, account.getDisplayName());
-        Assert.assertEquals("Email should match", TEST_EMAIL, account.getEmail());
-        Assert.assertEquals("Photo URL should match", TEST_PHOTO_URL, account.getPhotoUrl());
-        Assert.assertEquals("Thumbnail URL should match", TEST_THUMBNAIL_URL, account.getThumbnailUrl());
-        Assert.assertEquals("Language should match", TEST_LANGUAGE, account.getLanguage());
-        Assert.assertEquals("Locale should match", TEST_LOCALE, account.getLocale());
-        Assert.assertEquals("Additional OAuth values should match", createAdditionalOauthValues(), account.getAdditionalOauthValues());
-        Assert.assertEquals("Account name should match",
-                String.format("%s (%s) (%s)", TEST_USERNAME, TEST_INSTANCE_URL, SalesforceSDKManager.getInstance().getApplicationName()),
-                account.getAccountName());
+        checkTestAccount(account);
+    }
+
+    /**
+     * Tests populating account from token end point response and id response
+     */
+    @Test
+    public void testPopulateFromTokenEndpointAndIdService() throws JSONException {
```

Comment:
Test for new methods populateFromTokenEndpoint and populateFromIdServiceResponse